community/index/src/main/java/org/neo4j/index/gbptree/GBPTree.java

```diff
@@ -387,27 +497,99 @@ public void close() throws IOException
     @Override
     public IndexWriter<KEY,VALUE> writer( IndexWriter.Options options ) throws IOException
     {
-        SingleIndexWriter result = this.writer.getAndSet( null );
-        if ( result == null )
+        if ( !writerTaken.compareAndSet( false, true ) )
```

Comment:
`writerTaken` is a slightly different thing than the lock. We want an attempt to acquire multiple concurrent writers to fail, not to block and succeed when other writer as done. This to be certain that assumptions about users of this tree are correct.